### ISSUE TITLE
Support running Airbyte on Mac M1 Docker

### ISSUE THREAD

**Issue by jrhizor** [airbytehq/airbyte#2017]
## Current status
- Server can build and run.
- Can sync data among some connectors.
- Temporal can fail transiently and may require restarting the server to fix it.
- Some connectors do not work.

## Todos
- [ ] Fix temporal
- [ ] Fix connectors
  - [ ] `source-mssql`
  - [ ] `source-oracle`
  - [ ] `source-file`: stuck at `installTestReqs`
  - [ ] `source-mysql` & `destination-mysql`
    - MySQL connectors can compile after switching to MariaDB
    - There are weird unit test failures for `source-mysql`.

## Relevant PRs
- https://github.com/airbytehq/airbyte/pull/7205
- https://github.com/airbytehq/airbyte/pull/7208

**Comment by djKooks** [comment]
@jrhizor I'm also failing on Mac M1, with `./gradlew clean build`:

```
Build file '/Users/kination/workspace/oss/airbyte/airbyte-integrations/bases/airbyte-protocol/build.gradle' line: 3

An exception occurred applying plugin request [id: 'airbyte-docker']
> Failed to apply plugin 'airbyte-docker'.
...
```

**Comment by jrhizor** [comment]
Docker on M1 is now generally available https://docs.docker.com/docker-for-mac/apple-silicon/

**Comment by ashishgupta-97065** [comment]
Having the same issue on M1. Would be great to have a solution

**Comment by takan55** [comment]
Tried running airbyte with [Kind](https://kind.sigs.k8s.io/docs/user/quick-start/) on my M1 chip macbook pro but it got stuck at "cannot reach server. The server may still be starting up." forever. This is critical to me to stick with airbyte. Since more and more developers are switching to the M1 macs, I'd love to see the solution asap.

**Comment by tuliren** [comment]
M1 support reference: https://docs.docker.com/desktop/multi-arch/

**Comment by caiopedroso** [comment]
Hey guys! We are deciding on our primary connector, I'm advocating on airbyte, but all our engineers have an m1 at this moment, a possible solution to this would help a lot, do we have an eta?

**Comment by tuliren** [comment]
Hey @caiopedroso, thank you for letting us know. There is an increasing number of requests for making Airbyte work on M1. We are evaluating if we can prioritize this issue any time soon. Will provide an update soon (probably this Thursday).

**Comment by hbd** [comment]
Just confirming that running Airbyte on my M1 is not working. Short version of my experience:
1. The primary issue is uptime/reliability of the services. I have seen the Airbyte services successfully run a couple times, where I could access the dashboard or use the API. However, it always crashes within a few minutes of starting.
2. The `airbyte-temporal` service times out while trying `Waiting for default namespace to be initialized in temporal`. This happens about 10% of the time I start the services:
![image](https://user-images.githubusercontent.com/6836886/127408455-b92af324-d7df-435e-8e87-cc376c443bb9.png)
3. I am also seeing `airbyte-temporal` fail with `context deadline exceeded` errors when trying to access the webapp
```
airbyte-temporal   | {"level":"error","ts":"2021-07-27T21:49:54.515Z","msg":"Operation failed with internal error.","service":"matching","error":"UpdateTaskQueue failed. Failed to start t
ransaction. Error: context deadline exceeded","metric-scope":38,"logging-call-at":"persistenceMetricClients.go:804","stacktrace":"go.temporal.io/server/common/log/loggerimpl.(*loggerImpl).Error\n\t/temporal/common/log/loggerimpl/logger.go:138\ngo.temporal.io/server/common/persistence.(*taskPersistenceClient).updateErrorMetric\n\t/temporal/common/persistence/persistenceMetricClients.go:804\ngo.temporal.io/server/common/persistence.(*taskPersistenceClient).UpdateTaskQueue\n\t/temporal/common/persistence/persistenceMetricClients.go:787\ngo.temporal.io/server/service/matching.(*taskQueueDB).UpdateState\n\t/temporal/service/matching/db.go:109\ngo.temporal.io/server/service/matching.(*taskReader).persistAckLevel\n\t/temporal/service/matching/ta
skReader.go:272\ngo.temporal.io/server/service/matching.(*taskReader).getTasksPump\n\t/temporal/service/matching/taskReader.go:160"}
airbyte-temporal   | {"level":"error","ts":"2021-07-27T21:49:54.519Z","msg":"Persistent store operation failure","service":"matching","component":"matching-engine","wf-task-queue-name":"3218103eeed6:d20106e3-3a07-460c-8f98-6b1d259c7bd2","wf-task-queue-type":"Workflow","store-operation":"update-task-queue","error":"UpdateTaskQueue failed. Failed to start transaction. Error: context deadline exceeded","logging-call-at":"taskReader.go:166","stacktrace":"go.temporal.io/server/common/log/loggerimpl.(*loggerImpl).Error\n\t/temporal/common/log/loggerimpl/logger.
go:138\ngo.temporal.io/server/service/matching.(*taskReader).getTasksPump\n\t/temporal/service/matching/taskReader.go:166"}
airbyte-temporal   | {"level":"error","ts":"2021-07-27T21:50:00.405Z","msg":"Operation failed with internal error.","service":"matching","error":"GetMetadata operation failed. Error: cont
ext deadline exceeded","metric-scope":51,"logging-call-at":"persistenceMetricClients.go:932","stacktrace":"go.temporal.io/server/common/log/loggerimpl.(*loggerImpl).Error\n\t/temporal/common/log/loggerimpl/logger.go:138\ngo.temporal.io/server/common/persistence.(*metadataPersistenceClient).updateErrorMetric\n\t/temporal/common/persistence/persistenceMetricClients.go:932\ngo.temporal.io/server/common/persistence.(*metadataPersistenceClient).GetMetadata\n\t/temporal/common/persistence/persistenceMetricClients.go:910\ngo.temporal.io/server/common/cache.(*namespaceCache).refreshNamespacesLocked\n\t/temporal/common/cache/namespaceCache.go:435\ngo.temporal.io/server/common/cache.(*namespaceCache).refreshNamespaces\n\t/temporal/common/cache/name
spaceCache.go:425\ngo.temporal.io/server/common/cache.(*namespaceCache).refreshLoop\n\t/temporal/common/cache/namespaceCache.go:409"}
airbyte-temporal   | {"level":"error","ts":"2021-07-27T21:50:00.405Z","msg":"Error refreshing namespace cache","service":"matching","error":"GetMetadata operation failed. Error: context deadline exceeded","logging-call-at":"namespaceCache.go:414","stacktrace":"go.temporal.io/server/common/log/loggerimpl.(*loggerImpl).Error\n\t/temporal/common/log/loggerimpl/logger.go:138\ngo.temporal.io/server/common/cache.(*namespaceCache).refreshLoop\n\t/temporal/common/cache/namespaceCache.go:414"}
airbyte-temporal   | {"level":"error","ts":"2021-07-27T21:50:09.409Z","msg":"Operation failed with internal error.","service":"matching","error":"GetMetadata operation failed. Error: cont
ext deadline exceeded","metric-scope":51,"logging-call-at":"persistenceMetricClients.go:932","stacktrace":"go.temporal.io/server/common/log/loggerimpl.(*loggerImpl).Error\n\t/temporal/common/log/loggerimpl/logger.go:138\ngo.temporal.io/server/common/persistence.(*metadataPersistenceClient).updateErrorMetric\n\t/temporal/common/persistence/persistenceMetricClients.go:932\ngo.temporal.io/server/common/persistence.(*metadataPersistenceClient).GetMetadata\n\t/temporal/common/persistence/persistenceMetricClients.go:910\ngo.temporal.io/server/common/cache.(*namespaceCache).refreshNamespacesLocked\n\t/temporal/common/cache/namespaceCache.go:435\ngo.temporal.io/server/common/cache.(*namespaceCache).refreshNamespaces\n\t/temporal/common/cache/name
spaceCache.go:425\ngo.temporal.io/server/common/cache.(*namespaceCache).refreshLoop\n\t/temporal/common/cache/namespaceCache.go:409"}
airbyte-temporal   | {"level":"error","ts":"2021-07-27T21:50:09.410Z","msg":"Error refreshing namespace cache","service":"matching","error":"GetMetadata operation failed. Error: context deadline exceeded","logging-call-at":"namespaceCache.go:414","stacktrace":"go.temporal.io/server/common/log/loggerimpl.(*loggerImpl).Error\n\t/temporal/common/log/loggerimpl/logger.go:138\ngo.temporal.io/server/common/cache.(*namespaceCache).refreshLoop\n\t/temporal/common/cache/namespaceCache.go:414"}
```

I am moving on to workarounds like using a deployed version and my Intel. Happy to help however possible to get this up and running as soon as possible.

**Comment by tuliren** [comment]
> Hey @caiopedroso, thank you for letting us know. There is an increasing number of requests for making Airbyte work on M1. We are evaluating if we can prioritize this issue any time soon. Will provide an update this week (probably this Thursday).

Update on this. We have allocated one week for this issue in August, and will likely solve it by the end of the month.

**Comment by cgardens** [comment]
Another user affected by this problem: https://airbytehq.slack.com/archives/C01MFR03D5W/p1629153029204200

**Comment by bleonard** [comment]
That's me! (commenting for alerts when you fix it).

**Comment by barnyrelph** [comment]
Would love to see a fix for this too!

**Comment by vaibhavpals** [comment]
Having the same problem.

**Comment by KarlXerri** [comment]
+1 for this problem also.

**Comment by watkinsa** [comment]
+1, same problem running on M1.  I can get docker image to run and get into UI.  problem arises when trying to add a connector/source, even after pre-loading the connector manually and restarting.

**Comment by jwei98** [comment]
+1 following

**Comment by saadatqadri** [comment]
+1, ran into the same problem today as @hbd.

**Comment by ChristopheDuong** [comment]
It seems we'd also need to upgrade dbt version that is used by airbyte to run on Mac M1:
https://github.com/airbytehq/airbyte/issues/6872

**Comment by jrhizor** [comment]
For Temporal:
- https://github.com/temporalio/temporal/issues/1305
- https://github.com/airbytehq/airbyte/issues/7182

**Comment by tuliren** [comment]
Although we need to export env variable `JDK_VERSION=17` to run Airbyte on M1, the local JDK version should still on 14. Otherwise, task `:airbyte-db:jooq:generateConfigsDatabaseJooq` will run into the following issue:

```
2021-10-20 05:10:32 ERROR o.t.d.DockerClientProviderStrategy(lambda$getFirstValidStrategy$7):208 - {} - Could not find a valid Docker environment. Please check configuration. Attempted configurations were:
2021-10-20 05:10:32 ERROR o.t.d.DockerClientProviderStrategy(lambda$getFirstValidStrategy$7):210 - {} -     UnixSocketClientProviderStrategy: failed with exception RuntimeException (java.lang.UnsatisfiedLinkError: /Users/lirentu/Library/Caches/JNA/temp/jna3849754310316548901.tmp: dlopen(/Users/lirentu/Library/Caches/JNA/temp/jna3849754310316548901.tmp, 1): no suitable image found.  Did find:
        /Users/lirentu/Library/Caches/JNA/temp/jna3849754310316548901.tmp: no matching architecture in universal wrapper
        /Users/lirentu/Library/Caches/JNA/temp/jna3849754310316548901.tmp: no matching architecture in universal wrapper). Root cause UnsatisfiedLinkError (/Users/lirentu/Library/Caches/JNA/temp/jna3849754310316548901.tmp: dlopen(/Users/lirentu/Library/Caches/JNA/temp/jna3849754310316548901.tmp, 1): no suitable image found.  Did find:
        /Users/lirentu/Library/Caches/JNA/temp/jna3849754310316548901.tmp: no matching architecture in universal wrapper
        /Users/lirentu/Library/Caches/JNA/temp/jna3849754310316548901.tmp: no matching architecture in universal wrapper)
2021-10-20 05:10:32 ERROR o.t.d.DockerClientProviderStrategy(lambda$getFirstValidStrategy$7):210 - {} -     UnixSocketClientProviderStrategy: failed with exception RuntimeException (java.lang.NoClassDefFoundError: Could not initialize class com.github.dockerjava.transport.DomainSocket). Root cause NoClassDefFoundError (Could not initialize class com.github.dockerjava.transport.DomainSocket)
2021-10-20 05:10:32 ERROR o.t.d.DockerClientProviderStrategy(lambda$getFirstValidStrategy$7):212 - {} - As no valid configuration was found, execution cannot continue
2021-10-20 05:10:32 ERROR o.j.t.JooqLogger(error):454 - {} - Cannot read /Users/lirentu/git/airbyte/airbyte-db/jooq/build/tmp/generateConfigsDatabaseJooq/config.xml. Error : Failed to launch postgres container and run migration
java.lang.RuntimeException: Failed to launch postgres container and run migration
        at io.airbyte.db.instance.FlywayMigrationDatabase.getInternalConnection(FlywayMigrationDatabase.java:52) ~[io.airbyte.airbyte-db-lib-0.30.21-alpha.jar:?]
        at io.airbyte.db.instance.FlywayMigrationDatabase.create0(FlywayMigrationDatabase.java:44) ~[io.airbyte.airbyte-db-lib-0.30.21-alpha.jar:?]
        at org.jooq.meta.AbstractDatabase.create(AbstractDatabase.java:274) ~[jooq-meta-3.13.4.jar:?]
        at org.jooq.meta.AbstractDatabase.create(AbstractDatabase.java:263) ~[jooq-meta-3.13.4.jar:?]
        at org.jooq.meta.AbstractDatabase.setConnection(AbstractDatabase.java:253) ~[jooq-meta-3.13.4.jar:?]
        at org.jooq.codegen.GenerationTool.run0(GenerationTool.java:521) ~[jooq-codegen-3.13.4.jar:?]
        at org.jooq.codegen.GenerationTool.run(GenerationTool.java:222) ~[jooq-codegen-3.13.4.jar:?]
        at org.jooq.codegen.GenerationTool.generate(GenerationTool.java:217) ~[jooq-codegen-3.13.4.jar:?]
        at org.jooq.codegen.GenerationTool.main(GenerationTool.java:189) [jooq-codegen-3.13.4.jar:?]
```

which appears to be related to `testcontainers` prior to version 1.15.3, but we are already running 1.15.3 that is supposed to support M1.

**Comment by AnatolePledg** [comment]
Hi,

Trying to run Airbyte on M1 and encountering following error when trying to connect to a Postgres Source or Destination.

Is this related to this issue ?

Thanks

```
2021-11-10 10:29:40 INFO () TemporalAttemptExecution(lambda$getWorkerThread$2):170 - Completing future exceptionally...
io.airbyte.workers.WorkerException: Error while getting checking connection.
	at io.airbyte.workers.DefaultCheckConnectionWorker.run(DefaultCheckConnectionWorker.java:80) ~[io.airbyte-airbyte-workers-0.30.38-alpha.jar:?]
	at io.airbyte.workers.DefaultCheckConnectionWorker.run(DefaultCheckConnectionWorker.java:27) ~[io.airbyte-airbyte-workers-0.30.38-alpha.jar:?]
	at io.airbyte.workers.temporal.TemporalAttemptExecution.lambda$getWorkerThread$2(TemporalAttemptExecution.java:167) ~[io.airbyte-airbyte-workers-0.30.38-alpha.jar:?]
	at java.lang.Thread.run(Thread.java:832) [?:?]
Caused by: java.lang.RuntimeException: java.io.IOException: Cannot run program "/tmp/scripts7304386057438723395/image_exists.sh": error=0, Failed to exec spawn helper: pid: 149, exit value: 1
	at io.airbyte.workers.process.DockerProcessFactory.checkImageExists(DockerProcessFactory.java:155) ~[io.airbyte-airbyte-workers-0.30.38-alpha.jar:?]
	at io.airbyte.workers.process.DockerProcessFactory.create(DockerProcessFactory.java:76) ~[io.airbyte-airbyte-workers-0.30.38-alpha.jar:?]
	at io.airbyte.workers.process.AirbyteIntegrationLauncher.check(AirbyteIntegrationLauncher.java:67) ~[io.airbyte-airbyte-workers-0.30.38-alpha.jar:?]
	at io.airbyte.workers.DefaultCheckConnectionWorker.run(DefaultCheckConnectionWorker.java:49) ~[io.airbyte-airbyte-workers-0.30.38-alpha.jar:?]
	... 3 more
Caused by: java.io.IOException: Cannot run program "/tmp/scripts7304386057438723395/image_exists.sh": error=0, Failed to exec spawn helper: pid: 149, exit value: 1
	at java.lang.ProcessBuilder.start(ProcessBuilder.java:1128) ~[?:?]
	at java.lang.ProcessBuilder.start(ProcessBuilder.java:1071) ~[?:?]
	at io.airbyte.workers.process.DockerProcessFactory.checkImageExists(DockerProcessFactory.java:143) ~[io.airbyte-airbyte-workers-0.30.38-alpha.jar:?]
	at io.airbyte.workers.process.DockerProcessFactory.create(DockerProcessFactory.java:76) ~[io.airbyte-airbyte-workers-0.30.38-alpha.jar:?]
	at io.airbyte.workers.process.AirbyteIntegrationLauncher.check(AirbyteIntegrationLauncher.java:67) ~[io.airbyte-airbyte-workers-0.30.38-alpha.jar:?]
	at io.airbyte.workers.DefaultCheckConnectionWorker.run(DefaultCheckConnectionWorker.java:49) ~[io.airbyte-airbyte-workers-0.30.38-alpha.jar:?]
	... 3 more
Caused by: java.io.IOException: error=0, Failed to exec spawn helper: pid: 149, exit value: 1
	at java.lang.ProcessImpl.forkAndExec(Native Method) ~[?:?]
	at java.lang.ProcessImpl.<init>(ProcessImpl.java:319) ~[?:?]
	at java.lang.ProcessImpl.start(ProcessImpl.java:250) ~[?:?]
	at java.lang.ProcessBuilder.start(ProcessBuilder.java:1107) ~[?:?]
	at java.lang.ProcessBuilder.start(ProcessBuilder.java:1071) ~[?:?]
	at io.airbyte.workers.process.DockerProcessFactory.checkImageExists(DockerProcessFactory.java:143) ~[io.airbyte-airbyte-workers-0.30.38-alpha.jar:?]
	at io.airbyte.workers.process.DockerProcessFactory.create(DockerProcessFactory.java:76) ~[io.airbyte-airbyte-workers-0.30.38-alpha.jar:?]
	at io.airbyte.workers.process.AirbyteIntegrationLauncher.check(AirbyteIntegrationLauncher.java:67) ~[io.airbyte-airbyte-workers-0.30.38-alpha.jar:?]
	at io.airbyte.workers.DefaultCheckConnectionWorker.run(DefaultCheckConnectionWorker.java:49) ~[io.airbyte-airbyte-workers-0.30.38-alpha.jar:?]
```

**Comment by FilipeAraujo** [comment]
I'm having the same issue reported by @AnatolePledg. 

I was using an Intel based Mac and everything was working correctly, I moved to a new Mac M1 Pro, configured the same Postgres server and I'm having the reported issue.

I'm using the latest commit (cdb476ecad0b637efd4eb154a189a12117eab086).

**Comment by thompson0012** [comment]
i have the same issues here
i just moved to M1 pro macbook few days ago and feeling confused before i saw this issues

**Comment by h7kanna** [comment]
the above error is because of the airbyte worker not able to create a process on Mac M1

Add this to the worker environment in docker-compose and try once

JAVA_OPTS: "-Djdk.lang.Process.launchMechanism=vfork"

**Comment by thompson0012** [comment]
> 

![image](https://user-images.githubusercontent.com/51963680/143822084-8454294e-13a4-4b5f-9627-89c4037cb5fa.png)

the problem still exist when i add JAVA_OPTS in docker-compose yaml file

**Comment by hbd** [comment]
I tried running Airbyte on my M1 immediately after the fix was released, and things worked well. However, I just tried running for the first time since then and am seeing segfaults from temporal in latest (`v0.32.8-alpha`):
```
airbyte-temporal   | + + tctl cluster healthgrep
airbyte-temporal   | SERVING
airbyte-temporal   | temporal.api.workflowservice.v1.WorkflowService: SERVING
airbyte-temporal   | + echo 'Temporal server started.'
airbyte-temporal   | + '[' false '!=' true ']'
airbyte-temporal   | + register_default_namespace
airbyte-temporal   | + echo 'Registering default namespace: default.'
airbyte-temporal   | Temporal server started.
airbyte-temporal   | Registering default namespace: default.
airbyte-temporal   | + tctl --ns default namespace describe
airbyte-temporal   | qemu: uncaught target signal 11 (Segmentation fault) - core dumped
airbyte-temporal   | ./update-and-start-temporal.sh: line 86:   442 Segmentation fault      tctl --ns "${DEFAULT_NAMESPACE}" namespace describe
airbyte-temporal   | + echo 'Default namespace default not found. Creating...'
airbyte-temporal   | Default namespace default not found. Creating...
airbyte-temporal   | + tctl --ns default namespace register --rd 1 --desc 'Default namespace for Temporal Server.'
airbyte-worker     | 2021-11-30 00:13:32 WARN i.a.w.t.TemporalUtils(waitForTemporalServerAndLog):117 - {workspace_app_root=/tmp/workspace/scheduler/logs} - Waiting for default namespace to be initialized in temporal...
airbyte-temporal   | qemu: uncaught target signal 11 (Segmentation fault) - core dumped
airbyte-temporal   | Default namespace default registration complete.
airbyte-temporal   | Adding Custom*Field search attributes.
airbyte-temporal   | ./update-and-start-temporal.sh: line 86:   446 Segmentation fault      tctl --ns "${DEFAULT_NAMESPACE}" namespace register --rd "${DEFAULT_NAMESPACE_RETENTION}" --desc "Default namespace for Temporal Server."
airbyte-temporal   | + echo 'Default namespace default registration complete.'
airbyte-temporal   | + '[' '' '!=' true ']'
airbyte-temporal   | + add_custom_search_attributes
airbyte-temporal   | + echo 'Adding Custom*Field search attributes.'
airbyte-temporal   | + tctl --auto_confirm admin cluster add-search-attributes --name CustomKeywordField --type Keyword --name CustomStringField --type Text --name CustomTextField --type Text --name CustomIntField --type Int --name CustomDatetimeField --type Datetime --name Cus
tomDoubleField --type Double --name CustomBoolField --type Bool
airbyte-temporal   | qemu: uncaught target signal 11 (Segmentation fault) - core dumped
airbyte-temporal   | ./update-and-start-temporal.sh: line 97:   457 Segmentation fault      tctl --auto_confirm admin cluster add-search-attributes --name CustomKeywordField --type Keyword --name CustomStringField --type Text --name CustomTextField --type Text --name CustomIntF
```

**Comment by h7kanna** [comment]
@hbd you can try to build temporal on Mac M1 see if it stabilize the run

```
curl -OL https://github.com/temporalio/temporal/archive/refs/tags/v1.13.0.tar.gz
tar -xvf v1.13.0.tar.gz
cd temporal-1.13.0 && docker build . -t temporalio/auto-setup:1.13.0 --build-arg TARGET=auto-setup
```

**Comment by hbd** [comment]
> @hbd you can try to build temporal on Mac M1 to see if it stabilizes the run
> 
> ```
> curl -OL https://github.com/temporalio/temporal/archive/refs/tags/v1.13.0.tar.gz
> tar -xvf v1.13.0.tar.gz
> cd temporal-1.13.0 && docker build . -t temporalio/auto-setup:1.13.0 --build-arg TARGET=auto-setup
> ```

Thanks for the help @h7kanna. Unfortunately that has not fixed running Airbyte locally for me. I'm seeing error with `temporal`:
```
airbyte-scheduler  | 2021-12-06 04:27:17 INFO i.a.s.a.SchedulerApp(waitForServer):188 - {workspace_app_root=/tmp/workspace/scheduler/logs} - Waiting for server to become available...
airbyte-temporal   | {"level":"error","ts":"2021-12-06T04:27:17.723Z","msg":"Operation failed with internal error.","error":"GetTasks operation failed. Failed to get rows. Error: context deadline exceeded","metric-scope":35,"logging-call-
at":"persistenceMetricClients.go:1229","stacktrace":"go.temporal.io/server/common/log.(*zapLogger).Error\n\t/temporal/common/log/zap_logger.go:142\ngo.temporal.io/server/common/persistence.(*metricEmitter).updateErrorMetric\n\t/temporal/c
ommon/persistence/persistenceMetricClients.go:1229\ngo.temporal.io/server/common/persistence.(*taskPersistenceClient).GetTasks\n\t/temporal/common/persistence/persistenceMetricClients.go:645\ngo.temporal.io/server/service/matching.(*taskQ
ueueDB).GetTasks\n\t/temporal/service/matching/db.go:147\ngo.temporal.io/server/service/matching.(*taskReader).getTaskBatchWithRange.func1\n\t/temporal/service/matching/taskReader.go:184\ngo.temporal.io/server/service/matching.(*taskQueue
ManagerImpl).executeWithRetry.func1\n\t/temporal/service/matching/taskQueueManager.go:499\ngo.temporal.io/server/common/backoff.Retry.func1\n\t/temporal/common/backoff/retry.go:104\ngo.temporal.io/server/common/backoff.RetryContext\n\t/te
mporal/common/backoff/retry.go:125\ngo.temporal.io/server/common/backoff.Retry\n\t/temporal/common/backoff/retry.go:105\ngo.temporal.io/server/service/matching.(*taskQueueManagerImpl).executeWithRetry\n\t/temporal/service/matching/taskQue
ueManager.go:503\ngo.temporal.io/server/service/matching.(*taskReader).getTaskBatchWithRange\n\t/temporal/service/matching/taskReader.go:183\ngo.temporal.io/server/service/matching.(*taskReader).getTaskBatch\n\t/temporal/service/matching/
taskReader.go:206\ngo.temporal.io/server/service/matching.(*taskReader).getTasksPump\n\t/temporal/service/matching/taskReader.go:145\ngo.temporal.io/server/internal/goro.(*Group).Go.func1\n\t/temporal/internal/goro/group.go:57"}
```

Steps to reproduce:
1. Follow your instructions
2. `docker compose up -d`

To confirm that it was using the right image, I updated `docker-compose.yaml` to use `temporalio/auto-setup:1.13.0`:
```
CONTAINER ID   IMAGE                            COMMAND                  CREATED         STATUS         PORTS                                                                                         NAMES
fcd51f80e9d5   temporalio/auto-setup:1.13.0     "./entrypoint.sh aut…"   5 minutes ago   Up 5 minutes   6933-6935/tcp, 6939/tcp, 7234-7235/tcp, 7239/tcp, 0.0.0.0:7233->7233/tcp, :::7233->7233/tcp   airbyte-temporal
067c49fc1f0a   airbyte/scheduler:0.33.2-alpha   "/bin/bash -c ${APPL…"   5 minutes ago   Up 5 minutes                                                                                                 airbyte-scheduler
0ed2a70ff21e   airbyte/worker:0.33.2-alpha      "/bin/bash -c ${APPL…"   5 minutes ago   Up 5 minutes                                                                                                 airbyte-worker
b2b952a458e6   airbyte/db:0.33.2-alpha          "docker-entrypoint.s…"   5 minutes ago   Up 5 minutes   5432/tcp                                                                                      airbyte-db
1fecc8095cfc   airbyte/server:0.33.2-alpha      "/bin/bash -c ${APPL…"   5 minutes ago   Up 5 minutes   8000/tcp, 0.0.0.0:8001->8001/tcp, :::8001->8001/tcp                                           airbyte-server
20e10d9571f3   airbyte/webapp:0.33.2-alpha      "/docker-entrypoint.…"   5 minutes ago   Up 5 minutes   0.0.0.0:8000->80/tcp, :::8000->80/tcp                                                         airbyte-webapp
```

**Comment by longfellowone** [comment]
Recieving the following on 33.5-alpha. New M1 Macbook Pro

`https://github.com/longfellowone/airbyteerr.git`
`docker-compose up`

MEILI_MASTER_KEY=key

[logs-3b6db45f-3a7a-400a-89bd-7c3878270768-.txt](https://github.com/airbytehq/airbyte/files/7686068/logs-3b6db45f-3a7a-400a-89bd-7c3878270768-.txt)

[logs-734ead8b-74b0-4649-9f3e-99bf7b2dcd41-.txt](https://github.com/airbytehq/airbyte/files/7686069/logs-734ead8b-74b0-4649-9f3e-99bf7b2dcd41-.txt)

**Comment by pgdeepak91** [comment]
I'm also facing the same issue as @longfellowone said. when i try to connect to any db source or destination in macbook M1 air.

	at io.airbyte.workers.DefaultCheckConnectionWorker.run(DefaultCheckConnectionWorker.java:27) ~[io.airbyte-airbyte-workers-0.35.4-alpha.jar:?]
	at io.airbyte.workers.temporal.TemporalAttemptExecution.lambda$getWorkerThread$2(TemporalAttemptExecution.java:174) ~[io.airbyte-airbyte-workers-0.35.4-alpha.jar:?]
	at java.lang.Thread.run(Thread.java:833) [?:?]
Caused by: java.lang.RuntimeException: java.io.IOException: Cannot run program "/tmp/scripts7941208794739700256/image_exists.sh": error=0, Failed to exec spawn helper: pid: 130, exit value: 1
	at io.airbyte.workers.process.DockerProcessFactory.checkImageExists(DockerProcessFactory.java:199) ~[io.airbyte-airbyte-workers-0.35.4-alpha.jar:?]
	at io.airbyte.workers.process.DockerProcessFactory.create(DockerProcessFactory.java:96) ~[io.airbyte-airbyte-workers-0.35.4-alpha.jar:?]
	at io.airbyte.workers.process.AirbyteIntegrationLauncher.check(AirbyteIntegrationLauncher.java:60) ~[io.airbyte-airbyte-workers-0.35.4-alpha.jar:?]
	at io.airbyte.workers.DefaultCheckConnectionWorker.run(DefaultCheckConnectionWorker.java:53) ~[io.airbyte-airbyte-workers-0.35.4-alpha.jar:?]
	... 3 more
Caused by: java.io.IOException: Cannot run program "/tmp/scripts7941208794739700256/image_exists.sh": error=0, Failed to exec spawn helper: pid: 130, exit value: 1
	at java.lang.ProcessBuilder.start(ProcessBuilder.java:1143) ~[?:?]
	at java.lang.ProcessBuilder.start(ProcessBuilder.java:1073) ~[?:?]
	at io.airbyte.workers.process.DockerProcessFactory.checkImageExists(DockerProcessFactory.java:187) ~[io.airbyte-airbyte-workers-0.35.4-alpha.jar:?]
	at io.airbyte.workers.process.DockerProcessFactory.create(DockerProcessFactory.java:96) ~[io.airbyte-airbyte-workers-0.35.4-alpha.jar:?]
	at io.airbyte.workers.process.AirbyteIntegrationLauncher.check(AirbyteIntegrationLauncher.java:60) ~[io.airbyte-airbyte-workers-0.35.4-alpha.jar:?]
	at io.airbyte.workers.DefaultCheckConnectionWorker.run(DefaultCheckConnectionWorker.java:53) ~[io.airbyte-airbyte-workers-0.35.4-alpha.jar:?]
	... 3 more
	Caused by: java.io.IOException: error=0, Failed to exec spawn helper: pid: 130, exit value: 1
	at java.lang.ProcessImpl.forkAndExec(Native Method) ~[?:?]
	at java.lang.ProcessImpl.<init>(ProcessImpl.java:314) ~[?:?]
	at java.lang.ProcessImpl.start(ProcessImpl.java:244) ~[?:?]
	at java.lang.ProcessBuilder.start(ProcessBuilder.java:1110) ~[?:?]
	at java.lang.ProcessBuilder.start(ProcessBuilder.java:1073) ~[?:?]
	at io.airbyte.workers.process.DockerProcessFactory.checkImageExists(DockerProcessFactory.java:187) ~[io.airbyte-airbyte-workers-0.35.4-alpha.jar:?]
	at io.airbyte.workers.process.DockerProcessFactory.create(DockerProcessFactory.java:96) ~[io.airbyte-airbyte-workers-0.35.4-alpha.jar:?]
	at io.airbyte.workers.process.AirbyteIntegrationLauncher.check(AirbyteIntegrationLauncher.java:60) ~[io.airbyte-airbyte-workers-0.35.4-alpha.jar:?]
	at io.airbyte.workers.DefaultCheckConnectionWorker.run(DefaultCheckConnectionWorker.java:53) ~[io.airbyte-airbyte-workers-0.35.4-alpha.jar:?]
	... 3 more
		at io.temporal.internal.sync.POJOActivityTaskHandler$POJOActivityImplementation.execute(POJOActivityTaskHandler.java:252) ~[temporal-sdk-1.6.0.jar:?]
	at io.temporal.internal.sync.POJOActivityTaskHandler.handle(POJOActivityTaskHandler.java:209) ~[temporal-sdk-1.6.0.jar:?]
	at io.temporal.internal.worker.ActivityWorker$TaskHandlerImpl.handle(ActivityWorker.java:193) ~[temporal-sdk-1.6.0.jar:?]
	at io.temporal.internal.worker.ActivityWorker$TaskHandlerImpl.handle(ActivityWorker.java:151) ~[temporal-sdk-1.6.0.jar:?]
	at io.temporal.internal.worker.PollTaskExecutor.lambda$process$0(PollTaskExecutor.java:73) ~[temporal-sdk-1.6.0.jar:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136) [?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635) [?:?]
	at java.lang.Thread.run(Thread.java:833) [?:?]
Caused by: io.airbyte.workers.WorkerException: Error while getting checking connection.
	at io.airbyte.workers.DefaultCheckConnectionWorker.run(DefaultCheckConnectionWorker.java:84) ~[io.airbyte-airbyte-workers-0.35.4-alpha.jar:?]
	at io.airbyte.workers.DefaultCheckConnectionWorker.run(DefaultCheckConnectionWorker.java:27) ~[io.airbyte-airbyte-workers-0.35.4-alpha.jar:?]
	at io.airbyte.workers.temporal.TemporalAttemptExecution.lambda$getWorkerThread$2(TemporalAttemptExecution.java:174) ~[io.airbyte-airbyte-workers-0.35.4-alpha.jar:?]
	... 1 more
	Caused by: java.lang.RuntimeException: java.io.IOException: Cannot run program "/tmp/scripts7941208794739700256/image_exists.sh": error=0, Failed to exec spawn helper: pid: 130, exit value: 1
	at io.airbyte.workers.process.DockerProcessFactory.checkImageExists(DockerProcessFactory.java:199) ~[io.airbyte-airbyte-workers-0.35.4-alpha.jar:?]
	at io.airbyte.workers.process.DockerProcessFactory.create(DockerProcessFactory.java:96) ~[io.airbyte-airbyte-workers-0.35.4-alpha.jar:?]
	at io.airbyte.workers.process.AirbyteIntegrationLauncher.check(AirbyteIntegrationLauncher.java:60) ~[io.airbyte-airbyte-workers-0.35.4-alpha.jar:?]
	at io.airbyte.workers.DefaultCheckConnectionWorker.run(DefaultCheckConnectionWorker.java:53) ~[io.airbyte-airbyte-workers-0.35.4-alpha.jar:?]
	at io.airbyte.workers.DefaultCheckConnectionWorker.run(DefaultCheckConnectionWorker.java:27) ~[io.airbyte-airbyte-workers-0.35.4-alpha.jar:?]
	at io.airbyte.workers.temporal.TemporalAttemptExecution.lambda$getWorkerThread$2(TemporalAttemptExecution.java:174) ~[io.airbyte-airbyte-workers-0.35.4-alpha.jar:?]
	... 1 more
Caused by: java.io.IOException: Cannot run program "/tmp/scripts7941208794739700256/image_exists.sh": error=0, Failed to exec spawn helper: pid: 130, exit value: 1
	at java.lang.ProcessBuilder.start(ProcessBuilder.java:1143) ~[?:?]
	at java.lang.ProcessBuilder.start(ProcessBuilder.java:1073) ~[?:?]
	at io.airbyte.workers.process.DockerProcessFactory.checkImageExists(DockerProcessFactory.java:187) ~[io.airbyte-airbyte-workers-0.35.4-alpha.jar:?]
	at io.airbyte.workers.process.DockerProcessFactory.create(DockerProcessFactory.java:96) ~[io.airbyte-airbyte-workers-0.35.4-alpha.jar:?]
	at io.airbyte.workers.process.AirbyteIntegrationLauncher.check(AirbyteIntegrationLauncher.java:60) ~[io.airbyte-airbyte-workers-0.35.4-alpha.jar:?]
	at io.airbyte.workers.DefaultCheckConnectionWorker.run(DefaultCheckConnectionWorker.java:53) ~[io.airbyte-airbyte-workers-0.35.4-alpha.jar:?]
	at io.airbyte.workers.DefaultCheckConnectionWorker.run(DefaultCheckConnectionWorker.java:27) ~[io.airbyte-airbyte-workers-0.35.4-alpha.jar:?]
	at io.airbyte.workers.temporal.TemporalAttemptExecution.lambda$getWorkerThread$2(TemporalAttemptExecution.java:174) ~[io.airbyte-airbyte-workers-0.35.4-alpha.jar:?]

**Comment by pgdeepak91** [comment]
Caused by: java.io.IOException: error=0, Failed to exec spawn helper: pid: 130, exit value: 1
	at java.lang.ProcessImpl.forkAndExec(Native Method) ~[?:?]
	at java.lang.ProcessImpl.<init>(ProcessImpl.java:314) ~[?:?]
	at java.lang.ProcessImpl.start(ProcessImpl.java:244) ~[?:?]
	at java.lang.ProcessBuilder.start(ProcessBuilder.java:1110) ~[?:?]
	at java.lang.ProcessBuilder.start(ProcessBuilder.java:1073) ~[?:?]
	at io.airbyte.workers.process.DockerProcessFactory.checkImageExists(DockerProcessFactory.java:187) ~[io.airbyte-airbyte-workers-0.35.4-alpha.jar:?]
	at io.airbyte.workers.process.DockerProcessFactory.create(DockerProcessFactory.java:96) ~[io.airbyte-airbyte-workers-0.35.4-alpha.jar:?]
	at io.airbyte.workers.process.AirbyteIntegrationLauncher.check(AirbyteIntegrationLauncher.java:60) ~[io.airbyte-airbyte-workers-0.35.4-alpha.jar:?]
	at io.airbyte.workers.DefaultCheckConnectionWorker.run(DefaultCheckConnectionWorker.java:53) ~[io.airbyte-airbyte-workers-0.35.4-alpha.jar:?]
	at io.airbyte.workers.DefaultCheckConnectionWorker.run(DefaultCheckConnectionWorker.java:27) ~[io.airbyte-airbyte-workers-0.35.4-alpha.jar:?]
	at io.airbyte.workers.temporal.TemporalAttemptExecution.lambda$getWorkerThread$2(TemporalAttemptExecution.java:174) ~[io.airbyte-airbyte-workers-0.35.4-alpha.jar:?]
	... 1 more

**Comment by h7kanna** [comment]
The above problem is caused by Worker Java process not being able to run docker command. (Currently worker depends on docker CLI to create containers)

It is similar to https://stackoverflow.com/questions/61301818/java-failed-to-exec-spawn-helper-error-since-moving-to-java-14-on-linux

Try setting JAVA_OPTS: "-Djdk.lang.Process.launchMechanism=vfork" for the work and see if that works.

**Comment by jrhizor** [comment]
Just to update on the state of this whole issue:
- The platform can be built locally to work completely, passing all tests on M1.
- We still need to:
   1. Publish and test M1-compatible platform images
   2. Along the way, add something to the platform to indicate which connectors are supported. Maybe we'll only show M1-compatible connectors if you run Airbyte on M1 by default. We're not set on a specific approach for this yet. We'll definitely want to add this information to the docs, too.
   3. Start updating and publishing M1-compatible connectors.
   
1) slated for Jan or Feb
2) not sure when this will happen yet
3) likely won't get started until next quarter

**Comment by voidful** [comment]
I am able to build airbyte locally in M1 machine with a little bit more setting, just share what I have done here:



1. Install sdkman and open jdk  17
```
curl -s "https://get.sdkman.io" | bash
source "$HOME/.sdkman/bin/sdkman-init.sh"
sdk install java 17.0.2-open
```
2. Switching to python 3.8.12 via pyenv(.python-version)

3. Install airbyte(git clone and cd into the root)
```
export DOCKER_BUILD_PLATFORM=linux/arm64
export DOCKER_BUILD_ARCH=arm64
export JDK_VERSION=17
export NODE_VERSION=16.11.1
export JAVA_OPTS=-Djdk.lang.Process.launchMechanism=vfork
export LDFLAGS="-L/usr/local/opt/openssl/lib"
export CPPFLAGS="-I/usr/local/opt/openssl/include"
SUB_BUILD=PLATFORM ./gradlew build -x test
docker compose up 
```

**Comment by sw-yx** [comment]
@jrhizor  just an update on the Temporal side -  v1.14.2 onwards should be fine again. only v1.12 to v1.14 were affected by this M1 issue.

**Comment by reubano** [comment]
For those using macports

```bash
cd airbyte
git fetch

export DOCKER_BUILD_PLATFORM=linux/arm64
export DOCKER_BUILD_ARCH=arm64
export ALPINE_IMAGE=arm64v8/alpine:3.14
export POSTGRES_IMAGE=arm64v8/postgres:13-alpine
export JDK_VERSION=17
export JAVA_OPTS=-Djdk.lang.Process.launchMechanism=vfork
export LDFLAGS="-L/opt/local/lib -L/opt/local/libexec/openssl3/lib"
export CPPFLAGS="-I/opt/local/include -I/opt/local/libexec/openssl3/include"

docker compose up 
```

**Comment by grishick** [comment]
> the above error is because of the airbyte worker not able to create a process on Mac M1
> 
> Add this to the worker environment in docker-compose and try once
> 
> JAVA_OPTS: "-Djdk.lang.Process.launchMechanism=vfork"

This seems to be causing a new problem. Gradle fails with `vfork is not a supported process launch mechanism on this platform`

**Comment by tino** [comment]
At version 0.35.60-alpha I manage to get it up with:
```bash
export DOCKER_BUILD_PLATFORM=linux/arm64
export DOCKER_BUILD_ARCH=arm64
export ALPINE_IMAGE=arm64v8/alpine:3.14
export POSTGRES_IMAGE=arm64v8/postgres:13-alpine
export JDK_VERSION=17
VERSION=0.35.60-alpha docker-compose up
```
however that crashes:

```
airbyte-server      | Mar 28, 2022 8:11:38 PM org.glassfish.jersey.server.wadl.WadlFeature configure
airbyte-server      | WARNING: JAXBContext implementation could not be found. WADL feature is disabled.
airbyte-server      | 2022-03-28 20:11:41 INFO o.h.v.i.u.Version(<clinit>):21 - HV000001: Hibernate Validator 6.1.2.Final
airbyte-server      | 2022-03-28 20:11:43 INFO o.e.j.s.h.ContextHandler(doStart):860 - Started o.e.j.s.ServletContextHandler@5ebe903a{/,null,AVAILABLE}
airbyte-server      | 2022-03-28 20:11:43 INFO o.e.j.s.AbstractConnector(doStart):331 - Started ServerConnector@71531dd7{HTTP/1.1, (http/1.1)}{0.0.0.0:8001}
airbyte-server      | 2022-03-28 20:11:43 INFO o.e.j.s.Server(doStart):400 - Started @75367ms
airbyte-server      | 2022-03-28 20:11:43 INFO i.a.s.ServerApp(start):124 -
airbyte-server      |     ___    _      __          __
airbyte-server      |    /   |  (_)____/ /_  __  __/ /____
airbyte-server      |   / /| | / / ___/ __ \/ / / / __/ _ \
airbyte-server      |  / ___ |/ / /  / /_/ / /_/ / /_/  __/
airbyte-server      | /_/  |_/_/_/  /_.___/\__, /\__/\___/
airbyte-server      |                     /____/
airbyte-server      | --------------------------------------
airbyte-server      |  Now ready at http://localhost:8000/
airbyte-server      | --------------------------------------
airbyte-server      | Version: 0.35.60-alpha
airbyte-server      |
airbyte-server      | #
airbyte-server      | # A fatal error has been detected by the Java Runtime Environment:
airbyte-server      | #
airbyte-server      | #  SIGSEGV (0xb) at pc=0x00000040132256a0, pid=1, tid=87
airbyte-server      | #
airbyte-server      | # JRE version: OpenJDK Runtime Environment (17.0.1+12) (build 17.0.1+12-39)
airbyte-server      | # Java VM: OpenJDK 64-Bit Server VM (17.0.1+12-39, mixed mode, sharing, tiered, compressed oops, compressed class ptrs, g1 gc, linux-amd64)
airbyte-server      | # Problematic frame:
airbyte-server      | # J 5389 c1 com.ibm.icu.text.UnicodeSet.checkFrozen()V (19 bytes) @ 0x00000040132256a0 [0x00000040132255e0+0x00000000000000c0]
airbyte-server      | #
airbyte-server      | # No core dump will be written. Core dumps have been disabled. To enable core dumping, try "ulimit -c unlimited" before starting Java again
airbyte-server      | #
airbyte-server      | # An error report file with more information is saved as:
airbyte-server      | # /app/hs_err_pid1.log
airbyte-server      | #
airbyte-server      | # If you would like to submit a bug report, please visit:
airbyte-server      | #   https://bugreport.java.com/bugreport/crash.jsp
airbyte-server      | #
airbyte-server      | qemu: uncaught target signal 6 (Aborted) - core dumped
```

**Comment by harshitasingh1612** [comment]
Facing the same issue in Airbyte while trying to run a sample dataset using File connector on M1 Macbook Pro.Tried the different workarounds but still the same.



	at java.lang.ProcessBuilder.start(ProcessBuilder.java:1073) ~[?:?]
	at io.airbyte.workers.process.DockerProcessFactory.checkImageExists(DockerProcessFactory.java:174) ~[io.airbyte-airbyte-workers-0.35.66-alpha.jar:?]
	at io.airbyte.workers.process.DockerProcessFactory.create(DockerProcessFactory.java:94) ~[io.airbyte-airbyte-workers-0.35.66-alpha.jar:?]
	at io.airbyte.workers.process.AirbyteIntegrationLauncher.check(AirbyteIntegrationLauncher.java:58) ~[io.airbyte-airbyte-workers-0.35.66-alpha.jar:?]
	at io.airbyte.workers.DefaultCheckConnectionWorker.run(DefaultCheckConnectionWorker.java:53) ~[io.airbyte-airbyte-workers-0.35.66-alpha.jar:?]
	at io.airbyte.workers.DefaultCheckConnectionWorker.run(DefaultCheckConnectionWorker.java:27) ~[io.airbyte-airbyte-workers-0.35.66-alpha.jar:?]
	at io.airbyte.workers.temporal.TemporalAttemptExecution.lambda$getWorkerThread$2(TemporalAttemptExecution.java:155) ~[io.airbyte-airbyte-workers-0.35.66-alpha.jar:?]
	... 1 more
Caused by: java.io.IOException: error=0, Failed to exec spawn helper: pid: 147, exit value: 1
	at java.lang.ProcessImpl.forkAndExec(Native Method) ~[?:?]
	at java.lang.ProcessImpl.<init>(ProcessImpl.java:314) ~[?:?]
	at java.lang.ProcessImpl.start(ProcessImpl.java:244) ~[?:?]
	at java.lang.ProcessBuilder.start(ProcessBuilder.java:1110) ~[?:?]
	at java.lang.ProcessBuilder.start(ProcessBuilder.java:1073) ~[?:?]
	at io.airbyte.workers.process.DockerProcessFactory.checkImageExists(DockerProcessFactory.java:174) ~[io.airbyte-airbyte-workers-0.35.66-alpha.jar:?]
	at io.airbyte.workers.process.DockerProcessFactory.create(DockerProcessFactory.java:94) ~[io.airbyte-airbyte-workers-0.35.66-alpha.jar:?]
	at io.airbyte.workers.process.AirbyteIntegrationLauncher.check(AirbyteIntegrationLauncher.java:58) ~[io.airbyte-airbyte-workers-0.35.66-alpha.jar:?]
	at io.airbyte.workers.DefaultCheckConnectionWorker.run(DefaultCheckConnectionWorker.java:53) ~[io.airbyte-airbyte-workers-0.35.66-alpha.jar:?]
	at io.airbyte.workers.DefaultCheckConnectionWorker.run(DefaultCheckConnectionWorker.java:27) ~[io.airbyte-airbyte-workers-0.35.66-alpha.jar:?]
	at io.airbyte.workers.temporal.TemporalAttemptExecution.lambda$getWorkerThread$2(TemporalAttemptExecution.java:155) ~[io.airbyte-airbyte-workers-0.35.66-alpha.jar:?]

**Comment by shubhransh-locale** [comment]
I am trying to create simple source and destination 
Sources: Mysql, Pokimon API
Destination: Local CSV
always getting following error
`i.a.c.i.LineGobbler(voidCall):82 - WARNING: The requested image's platform (linux/amd64) does not match the detected host platform (linux/arm64/v8) and no specific platform was requested`

Is there a way to fix this?

**Comment by tovbinm** [comment]
Is there an ETA on this?

**Comment by Inventrohyder** [comment]
For me AirByte builds on an M1 and I don’t get any errors

however it doesn’t connect to any source or destination successfully

**Comment by tovbinm** [comment]
In the meantime, we compiled and published Airbyte containers for Apple M1 chips. Published on [farosai org on DockerHub](https://hub.docker.com/u/farosai). E.g. `airbyte/init` was published under `farosai/airbyte-init` and the same for all other Airbyte containers.

It would be great to see the official images published as well though.

**Comment by davinchia** [comment]
HI all, we've started publishing official ARM images as of 0.38.2-alpha. A proper announcement will go out soon. @evantahler and @sherifnada are working on publishing ARM connector images.

**Comment by evantahler** [comment]
With https://github.com/airbytehq/airbyte/pull/13004, we now have the plumbing in place to start releasing connectors for multiple platforms (e.g. M1 macs).  Going forward, we will start doing just that.

**Comment by erikcw** [comment]
Looks like `alpine/socat:1.7.4.1-r1` (`source-s3-check`) needs to be updated to version `1.7.4.3-r0` for arm support.

**Comment by davinchia** [comment]
@erikcw socat was updated in https://github.com/airbytehq/airbyte/pull/13411.

I tested this locally on an M1 laptop and was able to run a sync using the source-s3.

<img width="1964" alt="Screenshot 2022-06-02 at 5 12 28 PM" src="https://user-images.githubusercontent.com/9772859/171601851-fec47109-4eec-47e6-a4c3-4ade107ae3c9.png">

**Comment by davinchia** [comment]
I am going to close this issue since the core Airbyte platform now supports M1. Note that though we've the default connector publishing now publishes M1 images, not all connectors have M1 images published yet. Our approach is to greedily publish M1-compatible connector images as connectors are updated.

Please open a new issue for M1 specific connector image or any suspected M1-specific bugs. Thanks again for the patience everyone!

**Comment by evantahler** [comment]
@void666 we don't have a list at the moment, but if you are curious you can look on dockerhub and check for the `linux/arm64` build architecture.  It's safe to assume that most connectors in the Generally Available release stage have been re-published with M1 support by now.  

<img width="1658" alt="Screen Shot 2022-09-22 at 9 03 31 AM" src="https://user-images.githubusercontent.com/303226/191768279-aa477a89-20ca-4408-aa00-b6f52a688c0c.png">
